apps/docs/components/search-component.tsx

```diff
@@ -1,43 +1,41 @@
 "use client"
 
-import { useEffect, useState } from "react"
-import { DocSearchModal } from "@docsearch/react"
+import { useCallback, useRef, useState } from "react"
+import { DocSearchButton, DocSearchModal, useDocSearchKeyboardEvents } from "@docsearch/react"
 import { SearchIcon } from "lucide-react"
 
 import { cn } from "@/lib/utils"
 import { Button } from "@/registry/default/ui/button"
 
 export function Search() {
-  const [open, setOpen] = useState(false)
-
-  useEffect(() => {
-    const down = (e: KeyboardEvent) => {
-      if (
-        (e.target instanceof HTMLElement && e.target.isContentEditable) ||
-        e.target instanceof HTMLInputElement ||
-        e.target instanceof HTMLTextAreaElement ||
-        e.target instanceof HTMLSelectElement
-      ) {
-        return
-      }
-      if ((e.key === "k" && (e.metaKey || e.ctrlKey)) || e.key === "/") {
-        e.preventDefault()
-        setOpen((open) => !open)
-      }
-    }
-
-    document.addEventListener("keydown", down)
-    return () => document.removeEventListener("keydown", down)
-  }, [])
+  const searchButtonRef = useRef(null)
+  const [isOpen, setIsOpen] = useState(false)
+
+  const onOpen = useCallback(() => {
+    setIsOpen(true)
+  }, [setIsOpen])
+
+  const onClose = useCallback(() => {
+    setIsOpen(false)
+  }, [setIsOpen])
+
+  useDocSearchKeyboardEvents({
+    isOpen,
+    onOpen,
+    onClose,
+    searchButtonRef,
+  })
 
   return (
     <>
+      <DocSearchButton className="hidden" ref={searchButtonRef} onClick={onOpen} />
```

Comment:
Why do we need the hidden button? Does passing ref to the custom styled one not work?